js/piwik.js

```diff
@@ -5756,12 +5759,45 @@ if (typeof window.Matomo !== 'object') {
              * This should be used when your website is only available under HTTPS
              * so that all tracking cookies are always sent over secure connection.
              *
+             * Warning: If your site is available under http and https,
+             * setting this might lead to duplicate or incomplete visits.
+             *
              * @param bool
              */
             this.setSecureCookie = function (enable) {
+                if(enable && location.protocol !== 'https:') {
+                    logConsoleError("Error in setSecureCookie: You cannot use `Secure` on http.");
+                    return;
+                }
                 configCookieIsSecure = enable;
             };
 
+            /**
+             * Set the SameSite attribute for cookies to a custom value.
+             * You might want to use this if your site is running in an iframe since
+             * then it will only be able to access the cookies if SameSite is set to 'None'.
+             *
+             *
+             * Warning: 
+             * Sets CookieIsSecure to true on None, because None will only work with Secure; cookies
+             * If your site is available under http and https,
+             * using "None" might lead to duplicate or incomplete visits.
+             *
+             * @param string either Lax, None or Strict
+             */
+            this.setCookieSameSite = function (sameSite) {
+                if (sameSite === 'none' || sameSite === 'lax' || sameSite === 'strict') {
```

Comment:
@fealXX could we do add a `sameSite = String(sameSite).toLowerCase()` as a first line? So it also works for `None` and `NONE` etc? then below `else if` could be changed to `else` like 

```js
this.setCookieSameSite = function (sameSite) {
    sameSite = String(sameSite).toLowerCase()
     if (sameSite === 'none' || sameSite === 'lax' || sameSite === 'strict') {
         sameSite = sameSite.charAt(0).toUpperCase() + sameSite.toLowerCase().slice(1);
    } else (sameSite !== 'None' && sameSite !== 'Lax' && sameSite !== 'Strict') {
                    logConsoleError('Ignored value for sameSite. Please use either Lax, None, or Strict.');
       return;
     }
    ....
```

(also added a `.toLowerCase()` where it converts the sameSite to upper case first then lower case

Reply:
actually I just realise now that the code below also supports already like lowercase and upper case first so could also ignore this change. It be only interesting if we wanted to support `NONE` etc but that's not needed.

Reply:
I do think that is a more elegant solution though.